Sources/Commands/SwiftTool.swift

```diff
@@ -203,6 +204,17 @@ public class SwiftTool<Options: ToolOptions> {
         if let workspace = _workspace {
             return workspace
         }
+        let envSearchPaths = Utility.getEnvSearchPaths(
+            pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
+
+        guard let gitLocation = Utility.lookupExecutablePath(filename: Git.tool, searchPaths: envSearchPaths) else {
+            throw SwiftToolError.missingGitDependency
+            // NOTE(heckj): differentiate error reporting between not finding it and it not being executable?
+        }
+        guard localFileSystem.isExecutableFile(gitLocation) else {
+            throw SwiftToolError.missingGitDependency
+        }
```

Comment:
We should place this check in Process class (just before calling posix_spawn) so every subprocess we spawn is checked and not make it specialised for git.